rcljava_common/cmake/Modules/JavaExtra.cmake

```diff
@@ -165,3 +165,42 @@ function(ament_add_junit_tests TARGET_NAME)
 
   add_dependencies(${TARGET_NAME} "${TARGET_NAME}_jar")
 endfunction()
+
+#
+# Creates a sources jar from list of source files
+#
+# :param OUTPUT_NAME: Name of Jar
+# :type OUTPUT_NAME: string
+# :param SOURCES: Sources files
+# :type SOURCES: String list with file names
+#
+# @public
+#
+function(add_source_jar _TARGET_NAME)
+
+  cmake_parse_arguments(_add_source_jar
+    ""
+    "OUTPUT_NAME"
+    "SOURCES"
+    ${ARGN}
+  )
+
```

Comment:
We should probably handle the case where OUTPUT_NAME is not given. Maybe it can default to the `"${_TARGET_NAME}.jar"`?

Reply:
Done